Fix: device id is the raw SHA256 hash, instead of a string.

dapperstout opened the pull request:
Made the BEP documentation more in line with what the code is doing, and what is described in the discovery documentation.

calmh commented:
Yeah, maybe this should be locked down. When I wrote the protocol, the intention was not to control how devices are named or authenticated, but for interoperability that maybe wise. I'm not too fond of pointing to syncthing docs here though...

dapperstout commented:
I copied the part about device id's from the discovery protocol description. But I can understand it if you do not want to lock it down in the BEP protocol.

I changed the wording to make it clear that the device id doesn't have to be a SHA256, and removed the reference to the Syncthing documentation.

calmh commented:
:+1:
